### DISCUSSION TITLE
2022-06-14のミーティングアジェンダ

### DISCUSSION THREAD

**Discussion by azu** [asciidwango/js-primer#1431] [Meeting]
2022-06-14のミーティングで話すことを書くIssue

**Comment by azu** [comment]
# ES2022対応のリリース日

GA 123rd meeting
22.06.2022 - 23.06.2022
https://www.ecma-international.org/about-ecma/meeting-calendar/

なので、ES2022のリリースは6/23ぐらいの予定

**Comment by azu** [comment]
## ES2022の対応範囲

とりあえずウェブ版のリリースノート(予定)
[feat!: ES2022のサポート by azu · Pull Request #1432 · asciidwango/js-primer](https://github.com/asciidwango/js-primer/pull/1432)
これベースになると思う。

**Comment by azu** [comment]
# どこまでやるか

- [ビット演算子 ](https://jsprimer.net/basic/operator/#bit-operator)をまるっと削りたい気持ちもある
- ES Modulesの章の書き直し
- [Array#sliceの説明について · Issue #1163 · asciidwango/js-primer](https://github.com/asciidwango/js-primer/issues/1163)
- 複雑な章の分解
	- Class
	- 非同期
- Node.js ESMの対応
- Node.js 16 LTS対応
	- おそらく10月ぐらい
- 付録のアップデート
	- [付録: 参考リンク集 · JavaScript Primer #jsprimer](https://jsprimer.net/appendix/links/)

**Reply by lacolaco** [reply]
> 複雑な章の分解

もともと複雑なうえに仕様が増えたのでどうするか

**Reply by azu** [reply]
```
$ node -v
$ npm -v
```

npm 8にしたい

**Reply by lacolaco** [reply]
Nodeのバージョン

https://jsprimer.net/use-case/setup-local-env/#install-nodejs

**Reply by azu** [reply]
ビット演算子は、ちょっと長いので削ろうと思えば削れる

**Reply by lacolaco** [reply]
https://github.com/asciidwango/js-primer/issues/1337

**Reply by azu** [reply]
- Error Cause - メッセージを変えながらスタックトレースを維持する

**Reply by azu** [reply]
Node.js ESMの対応

- ユースケースCLIでやるには
  - Top-Level await → ESMで書く必要がある

**Reply by azu** [reply]
クラスの章を改訂するにあたって意見ある人を募集する

**Reply by azu** [reply]
クラスと非同期の改善案を考えるDiscussionを募集する。

**Reply by azu** [reply]
> Atom
> [Atom](https://atom.io/)はGitHub社がオープンソースで開発している無料のコードエディターです。 VSCodeと同じようにプラグインによる拡張性が高く、GitHubと連携した機能が特徴です。

は消す?

**Reply by azu** [reply]
オンライン

- CodeSandbox
- stackbliz
- Repl.it

**Reply by azu** [reply]
https://blog.glitch.com/post/an-exciting-leap-forward-for-glitch ホスティング

**Reply by azu** [reply]
モジュールバンドラー
esbuild

**Reply by azu** [reply]
Netlify
Vercel

**Comment by lacolaco** [comment]
sandpackは結局 https://github.com/asciidwango/js-primer/pull/1423#issuecomment-1143521776 の方針で、externalResourcesで対応して進めたほうがよいのか

**Reply by azu** [reply]
externalResourcesぐらいしか現状は手段なさそうな感じだった。

**Reply by lacolaco** [reply]
了解です

**Comment by azu** [comment]
NEXT

- Issue
- disussion
- リリースノート 改訂版に向けた作業を進めようとしてる
- レビューアー募集

**Reply by azu** [reply]
リリースノートはES2022のリリースと合わせる
